dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -1392,6 +1392,26 @@ def create_user(name, url: '/users.json', code: 201, **user_opts)
   end
 end
 
+And(/^I attempt to join the section$/) do
+  steps %Q{
+    Given I am on "#{@section_url}"
+  }
+end
+
+And(/^I fill in the sign up form with (in)?valid values for "([^"]*)"$/) do |invalid, name|
+  password = invalid ? 'Short' : 'ExtraLong'
+  email = "user#{Time.now.to_i}_#{rand(1_000_000)}@test.xx"
+  age = "10"
+  steps %Q{
+    And I type "#{name}" into "#user_name"
+    And I type "#{email}" into "#user_email"
+    And I type "#{password}" into "#user_password"
+    And I type "#{password}" into "#user_password_confirmation"
+    And I select the "#{age}" option in dropdown "user_age"
+    And I click selector ".btn.btn-primary" once I see it
```

Comment:
@drew-beckmen I would suggest trying one of the steps with the "to load a new page" clause Otherwise, it can lead to problems if the next step runs before the navigation happens.

Reply:
Thank you @davidsbailey!